### PR TITLE
Passing Resource Leak Checker test case

### PR DESCRIPTION
**Pull request by mernst**

### CODE REVIEW

**Review by kelloggm — DISMISSED** [review]

File: checker/tests/resourceleak/RlcThisTest.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  void test() {
+    Foo f1 = new Foo();
+    f1.b1(); // RLC reports a FP at this line
+    f1.a();
+
+    Foo f2 = new Foo();
+    f2.b2(); // RLC reports a FP at this line
+    f2.a();
+
+    f1 = new Foo();
```
Comment (msridhar) [inline checker/tests/resourceleak/RlcThisTest.java:32]: For clarity I'd prefer this to assign a new variable `f3`.  Also don't we get a false positive warning for this resource?  If so we could maybe add a comment

File: checker/tests/resourceleak/RlcThisTest.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    Foo ff1 = f1.b1();
+    ff1.a();
+
+    f2 = new Foo();
```
Comment (msridhar) [inline checker/tests/resourceleak/RlcThisTest.java:36]: Similarly I think this should assign a new variable `f4`, and add a comment if we are getting a false positive warning for this resource

**Review by msridhar — CHANGES_REQUESTED** [review]
Couple small changes if possible

File: checker/tests/resourceleak/RlcThisTest.java (hunk 1 above)
Comment (mernst) [reply checker/tests/resourceleak/RlcThisTest.java]: The error only occurs if there is a reassignment.  I have added versions of the test with and without the reassignment to show the difference in behavior.

**Review by msridhar — APPROVED** [review]

**Pull request merged**